pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -456,6 +458,75 @@ func (r *ReconcileClusterDeployment) reconcile(request reconcile.Request, cd *hi
 		return reconcile.Result{}, err
 	}
 
+	if cd.Spec.MachineManagementStrategy != nil && cd.Spec.MachineManagementStrategy.Strategy == "Central" {
+		targetNamespace := cd.Status.TargetNamespace
+		if targetNamespace == "" {
+			targetNamespace = apihelpers.GetResourceName(cd.Name+"-targetns", utilrand.String(5))
+		}
+
+		cdLog.Info("Creating the target namespace ", targetNamespace)
+		ns := &corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{Name: targetNamespace},
+		}
+		if err := r.Create(context.TODO(), ns); err != nil && !apierrors.IsAlreadyExists(err) {
```

Comment:
We probably should do a check it exists, these create mutating web calls should probably be avoided if possible, and a Get would be cached, so it seems like it would be ideal to just try the Get, if it doesn't exist Create.